cloudinit/net/__init__.py

```diff
@@ -327,8 +327,49 @@ def is_disabled_cfg(cfg):
     return cfg.get('config') == "disabled"
 
 
-def find_fallback_nic(blacklist_drivers=None):
+def find_fallback_nic_on(blacklist_drivers=None):
     """Return the name of the 'fallback' network device."""
+    if util.is_FreeBSD():
+        return find_fallback_nic_on_freebsd()
```

Comment:
Should these differences be captured on the distro class?

Reply:
i followed @goneri's example here, but, basically, probably, yes.

Reply:
The root of the problem is that most if the functions of `cloudinit/net/__init__.py` assume they are on Linux.

Reply:
from reading https://github.com/canonical/cloud-init/blob/master/cloudinit/distros/__init__.py it would appear this is the Linux os class